docs/development/custom.rst

```diff
@@ -0,0 +1,11 @@
+Customisation
+-------------
+
+.. toctree::
+   :maxdepth: 2
+
+   custom/extension
+   custom/command
+   custom/executor
+   custom/progress-logger
+   custom/report
```

Comment:
merge these two indxes - find a way to oranize this so the extension points are accessible from the navigation?